azure-pipelines.yml

```diff
@@ -28,19 +28,16 @@ jobs:
       py36-xdist:
         python.version: '3.6'
         tox.env: 'py36-xdist'
-      py37:
-        python.version: '3.7'
-        tox.env: 'py37'
-        # Coverage for:
-        # - _py36_windowsconsoleio_workaround (with py36+)
-        # - test_request_garbage (no xdist)
-        PYTEST_COVERAGE: '1'
       py37-linting/docs/doctesting:
         python.version: '3.7'
         tox.env: 'linting,docs,doctesting'
       py37-twisted/numpy:
         python.version: '3.7'
-        tox.env: 'py37-twisted,py37-numpy'
+        tox.env: 'py37-twisted-numpy'
```

Comment:
This PR looks good, but how about if we go further and just test `numpy` and `twisted` in the main `py37` job? It is after all just a matter of adding some extra dependencies and running a few extra tests, it seems it would be a win overall.

Reply:
That's what this does :) - but I will rename it.

Will also (re)move pexpect on Travis.

Reply:
> That's what this does :) - but I will rename it.

Oh indeed. 

But I meant to actually install `numpy` and `twisted` in all test envs as part of the normal build process (just like we install `mock` for all test envs).

Reply:
I think we tried this before, without much luck.

Also I do not like the added overhead of it.

Reply:
Which overhead you mean? I believe we will get a slight overhead during installation, but otherwise the extra tests should have a negligible cost.

Reply:
Yes, installation mostly.
In general I still do not like that our CI runs so much, but this PR makes it better, and also of course having dropped py27.. :)
I do not feel like we should slow it down again/more - but IIRC it is not even possible easily (since we have tried something like that).

Reply:
https://github.com/pytest-dev/pytest/pull/4849

Reply:
Thanks for the link. 👍